src/main/scala/Ring.scala

```diff
@@ -199,11 +199,22 @@ class TLRingNetwork(
   }
 }
 
-class RingSystemBus(params: SystemBusParams, buffer: TLNetworkBufferParams)
+case class RingSystemBusParams(params: SystemBusParams)
+  extends TLBusWrapperInstantiationLike {
+
+  def instantiate(context: HasTileLinkLocations, loc: Location[TLBusWrapper])(implicit p: Parameters): SystemBus = {
+    val ring = LazyModule(new RingSystemBus(params))
+    ring.suggestName(loc.name)
+    context.tlBusWrapperLocationMap += (loc -> ring)
+    ring
+  }
+}
+
+class RingSystemBus(params: SystemBusParams)
     (implicit p: Parameters) extends SystemBus(params) {
-  val system_bus_ring = LazyModule(new TLRingNetwork(buffer))
+  val system_bus_ring = LazyModule(new TLRingNetwork())
```

Comment:
Instead of getting rid of the `buffer` argument and always using the default defined in TLRingNetword, could you add it to RingSystemBusParams?